src/Utilities/VariableTemplates/VariableTemplates.jl

```diff
@@ -47,6 +47,20 @@ function varsindex(::Type{S}, insym::Symbol) where {S <: NamedTuple}
     end
     error("symbol '$insym' not found")
 end
+
+# return `Symbol`s unchanged.
+wrap_val(sym) = sym
+# wrap integer in `Val`
+wrap_val(i::Int) = Val(i)
+
+# We enforce that calls to `varsindex` on
+# an `NTuple` must be unrapped in `Val`.
```

Comment:
It'd be neater to enforce that everything be wrapped in a `Val` no? But this is not very important.

Reply:
It would, but then some code may work on the CPU but break on the GPU. This synchronizes CPU/GPU failures, which helped avoid a lot of debugging headaches.